tests/test_multipart_operations.py

```diff
@@ -4,92 +4,282 @@
+    def upload_multipart_object(
+        self,
+        c_scope_s3client,
+        tmp_directories_factory,
         amount=1,
-        min_size="20M",
-        max_size="30M",
-    )
-    # Upload multipart object
-    for i in range(len(object_names)):
-        get_upload_id = s3_client.initiate_multipart_object_upload(
-            bucket_name,
-            object_names[i],
+    ):
+        resp_dir = {}
+        origin_dir, results_dir = tmp_directories_factory(
+            dirs_to_create=["origin", "result"]
         )
-        all_part_info = []
-        file_name = origin_dir + "/" + object_names[i]
-        part_size = "10M"
-        part_data = split_file_data_for_multipart_upload(file_name, part_size)
-
-        for pd in range(len(part_data)):
-            part_id = pd + 1
-            part_info = s3_client.initiate_upload_part(
+        resp_dir["origin_dir"] = origin_dir
+        resp_dir["results_dir"] = results_dir
+        # 1. Create a bucket using S3
+        bucket_name = c_scope_s3client.create_bucket()
+        resp_dir["bucket_name"] = bucket_name
+        # 2. Write multipart objects to the bucket
+        object_names = generate_random_files(
+            origin_dir,
+            amount,
+            min_size="20M",
+            max_size="30M",
+        )
+        resp_dir["object_names"] = object_names
+        # Upload multipart object
+        log.info("Initiate multipart upload process")
+        for i in range(len(object_names)):
+            get_upload_id = c_scope_s3client.initiate_multipart_object_upload(
                 bucket_name,
                 object_names[i],
-                part_id,
-                get_upload_id,
-                part_data[pd],
             )
-            all_part_info.append({"PartNumber": part_id, "ETag": part_info["ETag"]})
-        list_mp_uploads = s3_client.list_multipart_upload(bucket_name)
-        log.info(list_mp_uploads)
-        mp_response = s3_client.complete_multipart_object_upload(
-            bucket_name, object_names[i], get_upload_id, all_part_info
+            resp_dir[f"{object_names[i]}_upload_id"] = get_upload_id
+            all_part_info = []
+            file_name = origin_dir + "/" + object_names[i]
+            part_size = "10M"
+            log.info(f"Split data into {part_size} size")
+            part_data = split_file_data_for_multipart_upload(file_name, part_size)
+            log.info("Initiate part uploads for multipart object")
+            for pd in range(len(part_data)):
+                part_id = pd + 1
+                part_info = c_scope_s3client.initiate_upload_part(
+                    bucket_name,
+                    object_names[i],
+                    part_id,
+                    get_upload_id,
+                    part_data[pd],
+                )
+                all_part_info.append({"PartNumber": part_id, "ETag": part_info["ETag"]})
+            resp_dir["all_part_info"] = all_part_info
+        return resp_dir
```

Comment:
This logic would be great to have available in other parts of the project - let's say for complex tests that incorporate multiple S3 features. Here are the modifications that would be needed to make that happen:

1. Move the method to a `bucket_utils.py` function (or whatever other name for the file you see fit)
2. Change the signature of the function to accept the name of a bucket and an S3Client instance
3. Move this bucket creation logic and the origin/results dir setup logic to a `setup` fixture in `test_multipart_operations.py` and use it in the tests

Reply:
Fixed